servlet/_index.md

```diff
@@ -1,7 +1,173 @@
+There are a number of methods that are automatically called upon via the `service()` method, and a servlet should implement those methods which provide the processing needed for the task at hand.
+
+* `doGet()`: handles `HTTP GET` requests for displaying data
+* `doPost()`: handles `HTTP POST` requests for performing inserts
+* `doPut()`: handles `HTTP PUT` requests for performing updates
+* `doDelete()`: handles `HTTP DELETE` requests for performing deletions
+* `doHead()`: handles `HTTP HEAD` requests
+* `doOptions()`: handles `HTTP OPTIONS` requests
+* `doTrace()`: handles `HTTP TRACE` requests
+
+In the case of loading some simple dynamic content, implementing the `doGet()` method may be all that is required. The `@WebServlet` annotation can be applied to a servlet class in order to register it with the container.
+The path to which a user must visit to invoke the servlet should be passed to the `@WebServlet` annotation.
+In the following example the servlet can be invoked by visiting the URL path containing the host name, port, and servlet path of `/HelloServlet` within a web browser:
+
+```Java
+@WebServlet("/HelloServlet")
+public class HelloServlet extends HttpServlet {
+    static String PAGE_HEADER = "<html><head><title>hello</title></head><body>";
+    static String PAGE_FOOTER = "</body></html>";
+    @Inject
+    HelloService service;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException,ServletException {
+        resp.setContentType("text/html");
+        PrintWriter writer = resp.getWriter();
+        writer.println(PAGE_HEADER);
+        try {
+            writer.println(service.printMessage("Jakarta EE!"));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        writer.println(PAGE_FOOTER);
+        writer.close();
+    }
+}
+```
+
+When the servlet URL is visited by a client, a request is sent to the host server, and then it hands off the request to the servlet container.
+The servlet container then determines which servlet to invoke and then calls upon it with the request and response objects.  
+
+The servlet life cycle is as follows:
+
+1. Loading and Instantiation: Servlet container loads the servlets that are registered and instantiates them for use.
+2. Initialization: `Servlet` initialization occurs calling upon the `init()` method of the servlet.
+3. RequestHandling: Request objects of type `ServletRequest` are handled, and then `ServletResponse` objects are filled out and passed as parameters to the `service()` method. At this stage, single requests or multiple asynchronous requests can be processed, depending upon configuration.
+4. End of Service: Once a container determines that a servlet is no longer needed, the `destroy()` method of the servlet is called to allow the servlet to release resources, if needed.
```

Comment:
This seems inaccurate and incredibly simplified for the Servlet spec.

Why document this here?
Isn't referencing the Servlet Spec itself more appropriate?

Off the top of my head, it looks like this ..

1. The Servlet Container loads / initializes.
2. Servlet Container is started
3. Request Handling
  a. Request received
  b. Servlet Container is selected by Context Path of Request
  c. Request is Dispatched to Servlet Container
  d. Handler Chain is determined based on url-patterns of Servlets (and optional Filters) and Dispatcher Type
  e. Handler Chain is called
  f. Eventually Servlet handles the request via the `service()` method.
  g. Response is produced
4. When Servlet Container is stopped, every Filter / Servlet / Listener that is interested in the stop can perform their stop duties
5. No more requests are dispatched to stopped Servlet Container.